metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py

```diff
@@ -685,34 +685,41 @@ def loop_tables(  # noqa: C901
         with data_reader or contextlib.nullcontext():
             try:
                 for table in inspector.get_table_names(schema):
-                    dataset_name = self.get_identifier(
-                        schema=schema, entity=table, inspector=inspector
-                    )
-
-                    if dataset_name not in tables_seen:
-                        tables_seen.add(dataset_name)
-                    else:
-                        logger.debug(
-                            f"{dataset_name} has already been seen, skipping..."
+                    try:
+                        dataset_name = self.get_identifier(
+                            schema=schema, entity=table, inspector=inspector
                         )
-                        continue
 
-                    self.report.report_entity_scanned(dataset_name, ent_type="table")
-                    if not sql_config.table_pattern.allowed(dataset_name):
-                        self.report.report_dropped(dataset_name)
-                        continue
+                        if dataset_name not in tables_seen:
+                            tables_seen.add(dataset_name)
+                        else:
+                            logger.debug(
+                                f"{dataset_name} has already been seen, skipping..."
+                            )
+                            continue
 
-                    try:
-                        yield from self._process_table(
-                            dataset_name,
-                            inspector,
-                            schema,
-                            table,
-                            sql_config,
-                            data_reader,
+                        self.report.report_entity_scanned(
+                            dataset_name, ent_type="table"
                         )
+                        if not sql_config.table_pattern.allowed(dataset_name):
+                            self.report.report_dropped(dataset_name)
+                            continue
+
+                        try:
```

Comment:
we already have a try here - can we just move the other code into this try instead of adding another try statement?

also, what errors were you seeing with these lines?

Reply:
Don't remember exactly but there was a scenario where if 1 of the table failed it would skip the whole Schema.